src/pykiso/lib/connectors/cc_serial.py

```diff
@@ -0,0 +1,174 @@
+        **kwargs
+    ):
+        """Init Serial settings
+
+        :param port: Device name (e.g. "COM1" for Windows or "/dev/ttyACM0" for Linux)
+        :param baudrate: Baud rate such as 9600 or 115200 etc, defaults to 9600
+        :param bytesize: Number of data bits. Possible values:
+          FIVEBITS, SIXBITS, SEVENBITS, EIGHTBITS, defaults to EIGHTBITS
+        :param parity: Enable parity checking. Possible values:
+          PARITY_NONE, PARITY_EVEN, PARITY_ODD PARITY_MARK, PARITY_SPACE,
+          defaults to PARITY_NONE
+        :param stopbits:  Number of stop bits. Possible values:
+          STOPBITS_ONE, STOPBITS_ONE_POINT_FIVE, STOPBITS_TWO,
+          defaults to STOPBITS_ONE
+        :param timeout: Set a read timeout value in seconds, defaults to None
+        :param xonxoff: Enable software flow contro, defaults to False
+        :param rtscts: Enable hardware (RTS/CTS) flow control, defaults to False
+        :param write_timeout: Set a write timeout value in seconds, defaults to None
+        :param dsrdtr: Enable hardware (DSR/DTR) flow control, defaults to False
+        :param inter_byte_timeout:  Inter-character timeout, None to disable,
+          defaults to None
+        :param exclusive: Set exclusive access mode (POSIX only).
+          A port cannot be opened in exclusive access mode if it is already open
+          in exclusive access mode., defaults to None
+        """
+
+        # Initialize the super class
+        super().__init__(**kwargs)
+        # Initialize the serial connection. Port None prevents automatic opening
+        self.serial = serial.Serial(
+            port=None,
+            baudrate=baudrate,
+            bytesize=bytesize,
+            parity=parity,
+            stopbits=stopbits,
+            timeout=timeout,
+            xonxoff=xonxoff,
+            rtscts=rtscts,
+            write_timeout=write_timeout,
+            dsrdtr=dsrdtr,
+            inter_byte_timeout=inter_byte_timeout,
+            exclusive=exclusive,
+        )
+
+        self.current_read_timeout = timeout
+        self.current_write_timeout = write_timeout
+        self.serial.port = port
+
+    def _cc_open(self) -> None:
+        """Open serial port"""
+        self.serial.open()
+
+    def _cc_close(self) -> None:
+        """Close serial port"""
+        self.serial.close()
+
+    def _cc_send(
+        self, msg: ByteString, raw: bool = True, timeout: float = None
+    ) -> None:
+        """Sends data to the serial port
+
+        :param msg: data to send
+        :param raw: unused
+        :param timeout: write timeout in seconds. None sets it to blocking,
+          defaults to None
+        :raises: SerialTimeoutException - In case a write timeout is configured
+          for the port and the time is exceeded.
+        """
+        if timeout != self.current_write_timeout:
+            self.current_write_timeout = timeout
+            self.serial.write_timeout = timeout
```

Comment:
```suggestion
        self.current_write_timeout = timeout or self.current_write_timeout
        self.serial.write_timeout = self.current_write_timeout
```
?

Reply:
The idea was to call self.serial.write_timeout only when necessary. -> reduce serial port reconfiguration
In your suggestion i would call the setter every time.